### PR TITLE
renaming AssertRequstMustHaveHeaders to AssertRequestMustIncludeHeaders

### LINKED ISSUE #7647: AssertRequestMustHaveHeaders doesn't work with empty headers
Specifically, `AssertRequestMustHaveHeaders` is only asserting the presence of any headers passed in. It cannot be used to assert the absence of headers right now, which is confusing given the name. I suggest renaming `AssertRequestMustHaveHeaders` to `AssertRequestMustIncludeHeaders`, asserting that the provided headers are not empty, and adding a new `AssertRequestMustHaveHeaders` that checks that the request headers exactly match the expected headers - ie. any keys present in one set of headers but not the other must cause a test failure.

Code: test/unit/net/http_loader.rs

### PR DESCRIPTION
**Pull request by psdh**
Fixes #7647

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7773)

### CODE REVIEW

File: tests/unit/net/http_loader.rs [hunk 1]
```diff
@@ -627,7 +647,7 @@ fn test_load_uses_explicit_accept_from_headers_in_load_data() {
     let hsts_list = Arc::new(RwLock::new(HSTSList::new()));
     let cookie_jar = Arc::new(RwLock::new(CookieStorage::new()));
 
-    let _ = load::<AssertRequestMustHaveHeaders>(load_data,
+    let _ = load::<AssertRequestMustIncludeHeaders>(load_data,
                                                  hsts_list,
```
Comment (Ms2ger) [inline tests/unit/net/http_loader.rs]: Adjust indentation.

**Comment by jdm** [comment]
Good start!
-S-awaiting-review +S-needs-code-changes -S-fails-tidy 


---

Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

<sup>**[tests/unit/net/http_loader.rs, line 136 \[r1\]](https://reviewable.io:443/reviews/servo/servo/7773#-K-ZZHY7a2TT8XfYayPn)** ([raw file](https://github.com/servo/servo/blob/4219d17164ddc48b6a8aa72f2460d6dbe87be0c5/tests/unit/net/http_loader.rs#L136)):</sup>
We should have two sets of headers like MustIncludeHeaders, since we need to verify that the actual headers set for this request exactly match the expected headers.


---

<sup>**[tests/unit/net/http_loader.rs, line 179 \[r1\]](https://reviewable.io:443/reviews/servo/servo/7773#-K-ZYxAk-cXrfgxDFSU6)** ([raw file](https://github.com/servo/servo/blob/4219d17164ddc48b6a8aa72f2460d6dbe87be0c5/tests/unit/net/http_loader.rs#L179)):</sup>
This is actually what AssertRequestMustHaveHeaders should be doing, since that should be an exact match. This method should be ensuring that `self.request_headers` includes all of the headers from `self.expected_headers`, but any others are fine to ignore.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7773)

**Comment by jdm** [comment]
-S-awaiting-review +S-needs-code-changes 


---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

<sup>**[tests/unit/net/http_loader.rs, line 155 \[r3\]](https://reviewable.io:443/reviews/servo/servo/7773#-K-dYxIjpvDJAt8-ud9j)** ([raw file](https://github.com/servo/servo/blob/a9d5bd7ba798b220fde5c931af30e60768a418fc/tests/unit/net/http_loader.rs#L155)):</sup>
This loop is unnecessary, given the next line.


---

<sup>**[tests/unit/net/http_loader.rs, line 155 \[r2\]](https://reviewable.io:443/reviews/servo/servo/7773#-K-dZ4TixhHvMFgvpzH7)** ([raw file](https://github.com/servo/servo/blob/878093ffe5f20426aa1012fce26b8a946d3168ad/tests/unit/net/http_loader.rs#L155)):</sup>
This looks like this doesn't compile now.


---

<sup>**[tests/unit/net/http_loader.rs, line 168 \[r3\]](https://reviewable.io:443/reviews/servo/servo/7773#-K-dZkuCosSWEBwbksWU)** ([raw file](https://github.com/servo/servo/blob/a9d5bd7ba798b220fde5c931af30e60768a418fc/tests/unit/net/http_loader.rs#L168)):</sup>
We should assert that expected_headers is not empty.


---

<sup>**[tests/unit/net/http_loader.rs, line 180 \[r3\]](https://reviewable.io:443/reviews/servo/servo/7773#-K-dZHF9gwgZyRJaGd6S)** ([raw file](https://github.com/servo/servo/blob/a9d5bd7ba798b220fde5c931af30e60768a418fc/tests/unit/net/http_loader.rs#L180)):</sup>
We no longer compare the actual values of the headers here, but we should.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7773)

**Comment by jdm** [comment]
As discussed, let's add a test for default HTTP headers that uses the new type.
-S-awaiting-review +S-needs-code-changes 


---

Reviewed 1 of 1 files at r4.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7773)

**Comment by jdm** [comment]
-S-awaiting-review +S-awaiting-answer 


---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[tests/unit/net/http_loader.rs, line 306 \[r5\]](https://reviewable.io:443/reviews/servo/servo/7773#-K-t7Y3n3Le4-140bDhw)** ([raw file](https://github.com/servo/servo/blob/1019f99a9d6e3a87f3282120062a9b8e037dab43/tests/unit/net/http_loader.rs#L306)):</sup>
Why post?


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7773)

**Comment by jdm** [comment]
Great! Could you squash these commits into one, please?
-S-awaiting-answer -S-awaiting-review +S-needs-squash 


---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7773)

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Pull request merged**